doc/nrf/libraries/modem/nrf_modem_lib.rst

```diff
@@ -101,6 +101,29 @@ If the modem trace medium is unable to keep up with the modem traces, the heap s
 Increasing the heap size allows more traces in the FIFO queue, but the trace heap will still be depleted if the modem continues to send traces at a rate faster than the rate at which the medium can handle over time.
 If increasing the trace heap size does not help, either optimize the medium speed or use a faster trace transport medium.
 
+Modem fault handling
+********************
+On initialization of the Modem library integration layer, it registers the :c:func:`nrf_modem_fault_handler` function through the initialization parameters.
+
+If a fault occurs in the modem, the application is notified through the fault handler.
+This enables the application to read the fault reason (in some cases the modem's program counter) and take appropriate action.
+
+The Modem library integration layer provides the following three options for :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT` Kconfig option:
+
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING` - This is the default Kconfig option that logs the Modem fault.
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_RESET_MODEM`- This Kconfig option restarts and reinitializes the modem and Modem library.
```

Comment:
What kind of "fault" could be failed over by this option? Does it include modem dump?

Reply:
Yes, all situations in which the modem core crashes for any reason

Reply:
Thanks, will enable this option for SLM app after libmodem v2.0 is in place